src/ImageLoaders/MzExe/NeImageLoader.cs

```diff
@@ -407,17 +420,45 @@ public override Program Load(Address addrLoad)
         {
             var cfgSvc = Services.RequireService<IConfigurationService>();
             this.arch = cfgSvc.GetArchitecture("x86-protected-16");
-            this.platform = cfgSvc.GetEnvironment("win16").Load(Services, arch);
             var rdr = new LeImageReader(RawImage, this.lfaNew);
             if (!LoadNeHeader(rdr))
                 throw new BadImageFormatException("Unable to read NE header.");
 
+            switch (bTargetOs)
+            {
+                case NE_TARGETOS.Windows:
+                case NE_TARGETOS.Windows386:
+                    this.platform = cfgSvc.GetEnvironment("win16").Load(Services, arch);
+                    break;
+                case NE_TARGETOS.EuropeanDos:
+                    this.platform = cfgSvc.GetEnvironment("ms-dos").Load(Services, arch);
+                    break;
+                default:
```

Comment:
Commit 027904892e1a4c5e8e95b61e194d4bf241597d4d added support for the `os2` platform, you can change this to:
```C#
case NE_TARGETOS.Os2:
    this.platform = cfgSvc.GetEnvironment("os2").Load(Services, arch);
    break;
```
if you have the time.